src/components/utils/src/logger/logger_impl.cc

```diff
@@ -71,9 +73,20 @@ void LoggerImpl::DeInit() {
 }
 
 void LoggerImpl::Flush() {
-  if (use_message_loop_thread_) {
-    if (loop_thread_) {
+  if (use_message_loop_thread_ && loop_thread_) {
+    assert(settings_);
+
+    if (0 == flush_logs_time_point_.time_since_epoch().count()) {
```

Comment:
why do we have such a condition? it seems impossible to me that count() is zero, thus impossible to execute next line

Reply:
Default constructor, creates a time_point representing the Clock's epoch (i.e., time_since_epoch() is zero).

Reply:
@VladSemenyuk in file on_exit_all_applications_notification.cc and function void OnExitAllApplicationsNotification::Run()
it calls SDL_INIT_FLUSH_LOGS_TIME_POINT(std::chrono::system_clock::now()); to set the value of flush_logs_time_point_ to non-zero, what's the condition that next line get executed? during the SDL running without receiving ExitAllApplicationsNotification?

Reply:
Yes, ExitAllApplicationsNotification is not the only exit point for SDL.

Reply:
@VladSemenyuk could you like what are other conditions or exit point to call logger Flush()?

Reply:
@yang1070 
When the following signals are thrown:
SIGINT
SIGTERM
SIGSEGV
or
"ASSERT" situation.

Reply:
@VladSemenyuk 
from the signals list and some internet search
The SIGINT (“program interrupt”) signal is sent when the user types the INTR character (normally C-c ).
the SIGSEGV is caused by segmentation fault
The SIGTERM signal is a generic signal used to cause program termination. Unlike SIGKILL , this signal can be blocked, handled, and ignored. It is the normal way to politely ask a program to terminate. The shell command kill generates SIGTERM by default.
ASSERT 

@VladSemenyuk it seems all other exit points (except IGNITION_OFF signal ) are not NORMAL exit points, and in those cases flush_logs_time_point_ will be zero block,  is calling loop_thread_->WaitDumpQueue() and wait all logs are written to disk a correct choice?
probably not. for example after segment fault, or after get a ctrl-c

Reply:
@VladSemenyuk  if without this change, loop_thread_->WaitDumpQueue() is called when condition (use_message_loop_thread_ && loop_thread_) is true, 
will  the signal  IGNITION_OFF cause SDL to call LoggerImpl::Flush(), thus get WaitDumpQueue get called?

if it is yes, then it is contrary to the following statement in the proposal 

In the current implementation, after receiving IGNITION_OFF signal, SDL drops all logs that have not yet been written. Such behavior sometimes prevents analyzing of SDL issues that are found by the ATF script, and requires adding extra timeouts in test scripts before SDL shutdown.

if it is no, in this change where is the logic or code that make sure after receiving IGNITION_OFF signal, LoggerImpl::Flush() is called?

Reply:
@yang1070 The proposal is old and at the time it was written the logger worked differently, and now it does`nt match the proposal. So before this implementation SDL flushed all logs always, and now configurable timeout shutdown, for IGNITION_OFF situation is added.

Reply:
@VladSemenyuk in that case I'm not sure the proposal is still valid and whether or not we need this change.

the proposal solves the problem that SDL drops off log and give options to continue write the log with maximum allowed time configured by the SDL settings in ini file.

Now (before this PR), SDL get some update, it always flush the log, SDL no long drop off the log, the old problem does not exist any more.  The problem may change to SDL takes too much time for flushing the logs? Why we still need to implement the proposal?

Reply:
> The problem may change to SDL takes too much time for flushing the logs? Why we still need to implement the proposal?

Yes. Problem is it takes  too much time for flushing the logs in some cases on weak environment.

Reply:
@VladSemenyuk in that case, I would like to see an update of SDL-117 and get the update approved first

Reply:
since the default SDL behavior is to flush log,
as in the code change, the setting `FlushLogMessagesBeforeShutdown` will only impact the normal IGNITION_OFF situation.  the name shall be change to something else that is more proper and can descript the situation